### PR TITLE
SDK: Enhance DotCMSLayout to Support GraphQL Response

### LINKED ISSUE #28737: SDK: Enhance DotCMSLayout to Support GraphQL Response
### Parent
#28404

### Task description
Right now, if you use our SDK to render pages, you can only use the page API to get the page data, either by fetching the URL directly or using our `@dotcms/client`.

The problem is that the Page API has a limitation in terms of how "deep" we can fetch relationship content, and some of our customers need more depth.

The good news is that our `GraphQL` API allows us to get all the relationships we want without any limitations, for example, we can have a query like this:

```graphql
query Test {
    page(url: "/testing-depth-with-gql") {
        title
        containers {
            containerContentlets {
                contentlets {
                    ... on Level1 {
                        title
                        level2 {
                            title
                            level3 {
                                title
                                level4 {
                                    title
                                    level5 {
                                        title
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

The problem is that the response from the page API is a little different from the one from GraphQL, for example:

In page API we have:

```json
"containers": {
  "//demo.dotcms.com/application/containers/default/": {
    "containerStructures": [],
    "rendered": [],
    "contentlets": {
      "uuid-10": [{...}],
      "uuid-1": [{...}]
    },
    "container": {...}
  }
}
```

While in GQL the same data will look more like this:

```json
{
    "containers": [
        {
            "identifier:": "//demo.dotcms.com/application/containers/default/",
            "containerContentlets": [
                {
                    "uuid": "uuid-1",
                    "contentlets": [{...}]
                },
                {
                    "uuid": "uuid-2",
                    "contentlets": [{...}]
                }
            ]
        }
    ]
}
```

Which, of course, if we pass the GQL one to `<DotcmsLayout /> it will not work.

### Acceptance Criteria
- The `<DotCMSLayout />` component should be updated to render data from page API or GraphQL
- Write a GQL query that includes all the needed page properties for `<DotCMSLayout />` to work correctly
- Update the nextjs example and add a new route specifically to use GQL, don't change our current catch-all route.

### External Links
- N/A

### Assumptions & Initiation Needs
- The developer should have knowledge of GraphQL queries and responses.
- Access to the DotCMS project where the DotCMSLayout component needs to be enhanced.

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates
N/A

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Let clients use `Graphql` to render `headless` pages.

### Video

https://github.com/dotCMS/core/assets/72418962/530a3f92-90e0-4ed7-9c4e-f25d36ef4d6a

### CODE REVIEW

File: examples/nextjs/src/app/campaigns/[[...slug]]/page.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { dotcmsClient, graphqlToPageEntity } from "@dotcms/client";
+import { MyPage } from "@/components/my-page";
+
+import { getGraphQLPageData } from "../../utils/gql";
+
+const client = dotcmsClient.init({
```
Comment (KevinDavilaDotCMS) [inline examples/nextjs/src/app/campaigns/[[...slug]]/page.js:6]: maybe we can centralize this in other file later

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: examples/nextjs/src/app/campaigns/[[...slug]]/page.js (hunk 1 above)
Comment (rjvelazco) [reply examples/nextjs/src/app/campaigns/[[...slug]]/page.js:6]: Thoughts @fmontes @zJaaal?

Comment (zJaaal) [reply examples/nextjs/src/app/campaigns/[[...slug]]/page.js:6]: I think I will need to centralize it for the content API Examples. Maybe is fine leave it here at the moment

File: core-web/yarn.lock
Comment (zJaaal) [inline core-web/yarn.lock:1]: Do we need to push this file?

**Review by zJaaal — APPROVED** [review]
Approved with one question.

File: core-web/yarn.lock
Comment (rjvelazco) [reply core-web/yarn.lock:1]: I'll revert this change.

File: examples/nextjs/src/app/globals.css [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+    overflow: hidden;
+  }
+
+  .line-clamp-3 {
+    display: -webkit-box;
+    -webkit-line-clamp: 3;
+    -webkit-box-orient: vertical;
+    overflow: hidden;
+  }
+}
```
Comment (fmontes) [inline examples/nextjs/src/app/globals.css:19]: Nice!

**Review by fmontes — APPROVED** [review]

**Pull request merged**